From 4a5bb36fb5f0e9e1e5bc1e8b97fbc5ab17bee7af Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 10 Apr 2014 17:33:25 +0200
Subject: [PATCH] extension-prefs: Force linking with libgnome-shell-js

Use the same hack we use for the main executable to fool the linker
when using --as-needed.

https://bugzilla.gnome.org/show_bug.cgi?id=727948
---
 src/gnome-shell-extension-prefs.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/gnome-shell-extension-prefs.c b/src/gnome-shell-extension-prefs.c
index 75f687e..e97df5d 100644
--- a/src/gnome-shell-extension-prefs.c
+++ b/src/gnome-shell-extension-prefs.c
@@ -50,3 +50,18 @@ main (int argc, char *argv[])
 
   return 0;
 }
+
+
+/* HACK:
+   Add a dummy function that calls into libgnome-shell-js.so to ensure it's
+   linked to /usr/bin/gnome-shell-extension-prefs even when linking with
+   --as-needed. This function is never actually called.
+   https://bugzilla.gnome.org/show_bug.cgi?id=670477
+*/
+void _shell_link_to_shell_js (void);
+
+void
+_shell_link_to_shell_js (void)
+{
+  shell_js_add_extension_importer (NULL, NULL, NULL, NULL);
+}
-- 
1.9.0
